src/EnergyPlus/WeatherManager.cc

```diff
@@ -3105,20 +3105,23 @@ namespace WeatherManager {
             if (!state.dataWeatherManager->LastHourSet) {
                 // For first day of weather, all time steps of the first hour will be
                 // equal to the first hour's value.
-                state.dataWeatherManager->LastHrOutDryBulbTemp = Wthr.OutDryBulbTemp(24);
-                state.dataWeatherManager->LastHrOutDewPointTemp = Wthr.OutDewPointTemp(24);
-                state.dataWeatherManager->LastHrOutBaroPress = Wthr.OutBaroPress(24);
-                state.dataWeatherManager->LastHrOutRelHum = Wthr.OutRelHum(24);
-                state.dataWeatherManager->LastHrWindSpeed = Wthr.WindSpeed(24);
-                state.dataWeatherManager->LastHrWindDir = Wthr.WindDir(24);
-                state.dataWeatherManager->LastHrSkyTemp = Wthr.SkyTemp(24);
-                state.dataWeatherManager->LastHrHorizIRSky = Wthr.HorizIRSky(24);
-                state.dataWeatherManager->LastHrBeamSolarRad = Wthr.BeamSolarRad(24);
-                state.dataWeatherManager->LastHrDifSolarRad = Wthr.DifSolarRad(24);
-                state.dataWeatherManager->LastHrAlbedo = Wthr.Albedo(24);
-                state.dataWeatherManager->LastHrLiquidPrecip = Wthr.LiquidPrecip(24);
-                state.dataWeatherManager->LastHrTotalSkyCover = Wthr.TotalSkyCover(24);
-                state.dataWeatherManager->LastHrOpaqueSkyCover = Wthr.OpaqueSkyCover(24);
+                // 2021-06: An additional input is added to here to allow the user to have chosen which hour to use
```

Comment:
Actual point where the Hr1 or Hr24 option is applied, depending on the processed input and environment settings.